capstone/src/main/webapp/login_script.js

```diff
@@ -6,31 +7,31 @@ function onSignIn(googleUser) {
 }
 
 function signOut() {
-    var auth2 = gapi.auth2.getAuthInstance();
-    auth2.signOut().then(function () {
-      console.log('User signed out.');
-    });
-    location.reload();
+  var auth2 = gapi.auth2.getAuthInstance();
+  auth2.signOut().then(function () {
+    console.log('User signed out.');
+  });
+  location.reload();
 }
 
 function isSignedIn() {
   return gapi.auth2.getAuthInstance().isSignedIn.get();
 }
 
 function checkLogin() {
-    if (!isSignedIn()) {
-        document.getElementById("report").style.display = "none";
-        document.getElementById("sign_out").style.display = "none";
-    } else {
-        document.getElementById("report").style.display = "block";
-        document.getElementById("sign_in").style.display = "none";
-    }
+  if (!isSignedIn()) {
+    document.getElementById("report").style.display = "none";
+    document.getElementById("sign_out").style.display = "none";
+  } else {
+    document.getElementById("report").style.display = "block";
+    document.getElementById("sign_in").style.display = "none";
+  }
 }
 
 function reload() {
-    setInterval(function() {
+  setInterval(function() {
     if(gapi.auth2.getAuthInstance().isSignedIn.get()) {
-        location.reload();
+      location.reload();
```

Comment:
What does this do?

Reply:
since the "report" button needs to be hidden when user isn't signed in and shown when user signs in, instead of having a listener always running, this script refreshes the page when user signs in to show the "report" button